frontend/src/components/custom-tools/prompt-card/OutputForIndex.jsx

```diff
@@ -1,31 +1,142 @@
 import PropTypes from "prop-types";
-import { Modal } from "antd";
+import { useState, useEffect, useRef, useCallback } from "react";
+import { Modal, Input, Button, Typography } from "antd";
 import "./PromptCard.css";
 import { uniqueId } from "lodash";
+import debounce from "lodash/debounce";
+import { ArrowDownOutlined, ArrowUpOutlined } from "@ant-design/icons";
 
 import { TextViewerPre } from "../text-viewer-pre/TextViewerPre";
 
 function OutputForIndex({ chunkData, setIsIndexOpen, isIndexOpen }) {
+  const [searchTerm, setSearchTerm] = useState("");
+  const [highlightedChunks, setHighlightedChunks] = useState([]);
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const activeRef = useRef(null);
+
+  // Split chunkData into chunks using <<< delimiter
+  const chunks = chunkData ? chunkData.split("\f\n") : [];
+  // To remove " at the end
+  if (chunks.length > 0) {
+    const lastChunk = chunks[chunks.length - 1].trim();
+    if (lastChunk === '\\n"' || lastChunk === "") {
+      chunks.pop();
+    }
+  }
```

Comment:
@jagadeeswaran-zipstack We should consider moving this logic inside the `useEffect` hook. Having it directly in the function will cause the logic to execute every time the component re-renders. A better approach would be to place it inside `useEffect`, allowing us to control when it re-executes by specifying dependencies.

Reply:
yeah make sense. I'll do it.